### PR TITLE
Document new date/time related functions.

### PR DESCRIPTION
**Pull request by brian-brazil**
@beorn7

### CODE REVIEW

File: content/docs/querying/functions.md [hunk 1]
```diff
@@ -325,6 +350,12 @@ expression is to be evaluated.
 
 `vector(s scalar)` returns the scalar `s` as a vector with no labels.
 
+## `year()`
+
+`year(v=vector(time()) instant-vector)` returns year
```
Comment (beorn7) [inline content/docs/querying/functions.md]: returns the year?

File: content/docs/querying/functions.md [hunk 2]
```diff
@@ -251,6 +271,11 @@ The special cases are equivalent to those in `ln`.
 `log10(v instant-vector)` calculates the decimal logarithm for all elements in `v`.
 The special cases are equivalent to those in `ln`.
 
+## `month()`
+
+`month(v=vector(time()) instant-vector)` returns the month of the year
+for each of the given times in UTC.
```
Comment (beorn7) [inline content/docs/querying/functions.md]: Just for clarity, perhaps add "January = 1 etc."

File: content/docs/querying/functions.md [hunk 3]
```diff
@@ -162,6 +177,11 @@ more trends in the data is considered. Both `sf` and `tf` must be between 0 and
 
 `holt_winters` should only be used with gauges.
 
+## `hour()`
+
+`hour(v=vector(time()) instant-vector)` returns the hour of the day
+for each of the given times in UTC.
```
Comment (beorn7) [inline content/docs/querying/functions.md]: Just for clarity, add "between 0 and 23".

File: content/docs/querying/functions.md [hunk 4]
```diff
@@ -63,6 +63,21 @@ vector as a scalar. This is in contrast to the `count()`
 always returns a vector (an empty one if the input vector is empty) and allows
 grouping by labels via a `by` clause.
 
+## `day_of_month()`
+
+`day_of_month(v=vector(time()) instant-vector)` returns the day of the month
+for each of the given times in UTC.
```
Comment (beorn7) [inline content/docs/querying/functions.md]: For clarity: "between 1 and 31"

File: content/docs/querying/functions.md (hunk 1 above)
Comment (brian-brazil) [inline content/docs/querying/functions.md]: Done

File: content/docs/querying/functions.md (hunk 2 above)
Comment (brian-brazil) [inline content/docs/querying/functions.md]: Done

**Comment by grobie** [comment]
The documentation of the default value might not be immediately clear. Maybe add an explicit sentence `If the parameted is omitted, the result of `time()` is used.` or so?

File: content/docs/querying/functions.md (hunk 3 above)
Comment (brian-brazil) [inline content/docs/querying/functions.md]: Done

File: content/docs/querying/functions.md (hunk 4 above)
Comment (brian-brazil) [inline content/docs/querying/functions.md]: Done

**Comment by beorn7** [comment]
Yeah, @grobie has a point. For a Python person, the documentation syntax is obvious, but perhaps not for everybody.

**Comment by grobie** [comment]
Or we describe the format at the top for non-python devs?

File: content/docs/querying/functions.md [hunk 5]
```diff
@@ -63,6 +63,21 @@ vector as a scalar. This is in contrast to the `count()`
...
 
+## `day_of_month()`
+
+`day_of_month(v=vector(time()) instant-vector)` returns the day of the month
+for each of the given times in UTC. Returned values are from 1 to 31.
+
+## `day_of_week()`
+
+`day_of_week(v=vector(time()) instant-vector)` returns the day of the week
+for each of the given times in UTC. Sunday=0 etc.
```
Comment (grobie) [inline content/docs/querying/functions.md]: For consistency, `Returned values are from 0 to 6 (Sunday=0)`?

**Comment by beorn7** [comment]
Yeah, let's have an explanation of default values at the top. We have a fair number now (although they are all time), but we might get more.

File: content/docs/querying/functions.md (hunk 5 above)
Comment (beorn7) [inline content/docs/querying/functions.md]: But then it should also be the same for the month... ;)

**Comment by brian-brazil** [comment]
That's updated. round also has a default.

**Comment by beorn7** [comment]
Nice.

> round also has a default.

Which settles our previous debate. :)

In different news: I have added a commit to make the `month()` and the `day_of_week()` wording more explicit and consistent, as @grobie suggested.

If you all like this, we can merge, and I'll hit the release button for 1.1.

**Comment by grobie** [comment]
Thanks everyone!

**Pull request merged**